lib/data/odata-filter.js

```diff
@@ -24,13 +24,16 @@ const odataFilter = (expr, odataToColumnMap) => {
   // I don't want to pass it to all of them.
 
   const extractFunctions = ['year', 'month', 'day', 'hour', 'minute', 'second'];
-  const methodCall = (fn, params) => {
+  const methodCall = (node) => {
     // n.b. odata-v4-parser appears to already validate function name and arity.
+    const fn = node.value.method;
     const lowerName = fn.toLowerCase();
+    const params = node.value.parameters;
     if (extractFunctions.includes(lowerName))
       return sql`extract(${raw(lowerName)} from ${op(params[0])})`; // eslint-disable-line no-use-before-define
     else if (fn === 'now')
       return sql`now()`;
+    else throw Problem.internal.unsupportedODataExpression({ at: node.position, type: node.type, text: node.raw });
```

Comment:
Would you mind putting the `throw` on its own line to parallel the `if` and `else if` above?

Reply:
I've done that, and added the missing brackets

Reply:
I actually don't mind the missing brackets, we often drop them if all the `if`/`else if`/`else` are one-liners. I'm also OK with brackets though, no need to remove them or anything.